File: 2 FORTRAN/Task_1/Task_1/fun_F_X.f90
real(kind=8) function fun_F_x(x)
    use point_para
    
    implicit none
    
    real(kind=8) :: X
    real(kind=8) :: epsilon_u
    character(len=3) :: typen
    Real(kind=8) ::  erf,sqrt,log
    intrinsic erf,sqrt,log
    
    epsilon_u = p_epu
    typen = p_typen
    
    if (typen=='III') then
        fun_F_x=.5-.5*erf((log(-X+1.0)+.5*log(2.0))/sqrt(2.0*log(2.0)))    
    end if
    
    return
end function

File: 3 Mathematica/20170613/Functions/Task_2/fun_F_x.m
% Jun 13, symbolic
% Updated in Mar18,2 017
% Task #20170302
% Related to May16/Oct17,2016
% edited in Feb21,2017
%%
function F_x =fun_F_x(x,typen,epsilon_u)

if strcmp(typen,'O')
    temp1 = (pi/4)/sin(pi/4)-x;
    temp2 = (pi/2)/sin(pi/2)-((pi/4)/sin(pi/4))^2;
    temp3 = 1+(temp1/sqrt(temp2))^(-4);
    F_x = 1-(temp3)^(-1);
elseif strcmp(typen,'I')
    temp = (x+epsilon_u)/(2*epsilon_u);
   F_x = heaviside(x+epsilon_u)*(temp)+(1-temp)*heaviside(x-epsilon_u);
elseif strcmp(typen,'II')
    temp=sqrt(1/pi)-x*sqrt(.5-1/pi);
    F_x =  1-erf(temp);
elseif strcmp(typen,'III')
    F_x=.5-.5*erf((log(-x+1)+.5*log(2))/sqrt(2*log(2)));
elseif strcmp(typen,'IV')
    F_x =((3-2*x)/sqrt(3))^(-3);      
end
return